piker/brokers/ib.py

```diff
@@ -1858,122 +1952,132 @@ async def stream_quotes(
+            pack in a `dict`-msg.
 
-        # TODO: more consistent field translation
-        atype = syminfo['asset_type'] = asset_type_map[syminfo['secType']]
+            '''
+            # pass back some symbol info like min_tick, trading_hours, etc.
+            syminfo = asdict(details)
+            syminfo.update(syminfo['contract'])
 
-        # for stocks it seems TWS reports too small a tick size
-        # such that you can't submit orders with that granularity?
-        min_tick = 0.01 if atype == 'stock' else 0
+            # nested dataclass we probably don't need and that won't IPC
+            # serialize
+            syminfo.pop('secIdList')
 
-        syminfo['price_tick_size'] = max(syminfo['minTick'], min_tick)
+            # TODO: more consistent field translation
+            atype = syminfo['asset_type'] = asset_type_map[syminfo['secType']]
 
-        # for "traditional" assets, volume is normally discreet, not a float
-        syminfo['lot_tick_size'] = 0.0
+            # for stocks it seems TWS reports too small a tick size
+            # such that you can't submit orders with that granularity?
+            min_tick = 0.01 if atype == 'stock' else 0
 
-        # TODO: for loop through all symbols passed in
-        init_msgs = {
-            # pass back token, and bool, signalling if we're the writer
-            # and that history has been written
-            sym: {
-                'symbol_info': syminfo,
-                'fqsn': first_quote['fqsn'],
-            }
-        }
-        return init_msgs
-
-    init_msgs = mk_init_msgs()
+            syminfo['price_tick_size'] = max(syminfo['minTick'], min_tick)
 
-    # TODO: we should instead spawn a task that waits on a feed to start
-    # and let it wait indefinitely..instead of this hard coded stuff.
-    with trio.move_on_after(1):
-        contract, first_ticker, details = await _trio_run_client_method(
-            method='get_quote',
-            symbol=sym,
-        )
+            # for "traditional" assets, volume is normally discreet, not
+            # a float
+            syminfo['lot_tick_size'] = 0.0
 
-    # it might be outside regular trading hours so see if we can at
-    # least grab history.
-    if isnan(first_ticker.last):
-        task_status.started((init_msgs, first_quote))
+            ibclient = proxy._aio_ns.ib.client
+            host, port = ibclient.host, ibclient.port
 
-        # it's not really live but this will unblock
-        # the brokerd feed task to tell the ui to update?
-        feed_is_live.set()
+            # TODO: for loop through all symbols passed in
+            init_msgs = {
+                # pass back token, and bool, signalling if we're the writer
+                # and that history has been written
+                sym: {
+                    'symbol_info': syminfo,
+                    'fqsn': first_quote['fqsn'],
+                },
+                'status': {
+                    'data_ep': f'{host}:{port}',
```

Comment:
Shows the preferred data client sockaddr in the feed status label in the sidepane.